### ISSUE TITLE
In the SSVAE code, when normalizing MNIST images, why not round to a binary image?

### ISSUE THREAD

**Issue by alanlisten** [pyro-ppl/pyro#1773]
In the SSVAE code，when normalizing MNIST images, why not round to a binary image? Because the likehood is a discrete Bernoulli distribution in the model function, and obs is a continuous picture on [0,1], there is an irreversible error.

**Comment by eb8680** [comment]
Hi, [`torch.distributions.Bernoulli.log_prob`](https://pytorch.org/docs/master/_modules/torch/distributions/bernoulli.html#Bernoulli.log_prob) uses [`torch.nn.BCEWithLogitsLoss`](https://pytorch.org/docs/master/nn.html#torch.nn.BCEWithLogitsLoss) under the hood, so it accepts values in the unit interval (try it in a repl!).  I agree it's a little odd, but this is a common enough use case for the Bernoulli distribution that it makes sense to support it.

Please ask questions on our [discussion forum](http://forum.pyro.ai/); we generally reserve GitHub issues for bug reports, feature requests, and other development discussion.  Questions or issues about PyTorch that are not Pyro-specific are more suited for PyTorch's forum or issue tracker and will get better responses there.
